### ISSUE TITLE
rbind,DataFrame-method fails when column is Date

### ISSUE THREAD

**Issue by PeteHaitch** [Bioconductor/S4Vectors#19]
``` r
suppressPackageStartupMessages(library(S4Vectors))
df1 <- DataFrame(x = 1, y = Sys.Date())
df2 <- DataFrame(x = 2, y = Sys.Date())
rbind(df1, df2)
#> Error in (function (classes, fdef, mtable) : unable to find an inherited method for function 'bindROWS' for signature '"Date"'
```

Created on 2018-06-25 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

```R
> BiocManager::valid()

* sessionInfo()

R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils    
[7] datasets  methods   base     

other attached packages:
[1] S4Vectors_0.19.15   BiocGenerics_0.27.1 repete_0.0.0.9011  
[4] devtools_1.13.5    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.17           lobstr_0.0.0.9000     
 [3] XVector_0.21.3         magrittr_1.5          
 [5] GenomicRanges_1.33.6   zlibbioc_1.27.0       
 [7] IRanges_2.15.14        munsell_0.5.0         
 [9] colorspace_1.3-2       rlang_0.2.1           
[11] stringr_1.3.1          GenomeInfoDb_1.17.1   
[13] plyr_1.8.4             tools_3.5.0           
[15] withr_2.1.2            fortunes_1.5-4        
[17] digest_0.6.15          GenomeInfoDbData_1.1.0
[19] BiocManager_1.30.1     pryr_0.1.4            
[21] bitops_1.0-6           codetools_0.2-15      
[23] RCurl_1.95-4.10        memoise_1.1.0         
[25] stringi_1.2.3          compiler_3.5.0        
[27] scales_0.5.0          

Bioconductor version '3.8'

  * 0 packages out-of-date
  * 1 packages too new

create a valid installation with

  BiocManager::install("bsseq", update = TRUE, ask = FALSE)

more details: BiocManager::valid()$too_new, BiocManager::valid()$out_of_date

Warning message:
0 packages out-of-date; 1 packages too new 
```

**Comment by hpages** [comment]
Looks like a regression I introduced when I refactored the `rbind` method for DataFrame objects (commit 1664d3f14e811324d462f25ced935a90764b7758), sorry.

Should be fixed in S4Vectors 0.19.17 (de1ddd02744f8f5514cdf27c89fd53d02e342187).

**Comment by PeteHaitch** [comment]
Thanks!
